crates/bls-gadgets/src/hash_to_group.rs

```diff
@@ -233,149 +224,135 @@ pub fn hash_to_bits<F: PrimeField, CS: ConstraintSystem<F>>(
         } else {
             let message = message
                 .iter()
-                .map(|m| m.get_value().get())
+                .map(|m| m.value())
                 .collect::<Result<Vec<_>, _>>()?;
             let message = bits_to_bytes(&message);
             let hash_result = DirectHasher.xof(&personalization, &message, 64).unwrap();
-            let mut bits = bytes_to_bits(&hash_result, 512);
-            bits.reverse();
+            let bits = bytes_to_bits(&hash_result, 512);
             bits
         };
-        constrain_bool(&mut cs, &bits)?
+
+        bits.iter()
+        .enumerate()
+        .map(|(_j, b)| Boolean::new_witness(message[..].cs(), || Ok(b)))
+        .collect::<Result<Vec<_>, _>>()?
     };
 
     Ok(xof_bits)
 }
 
-impl<P: Bls12Parameters> HashToGroupGadget<P> {
+impl<P: Bls12Parameters> HashToGroupGadget<P, Bls12_377_Fq> {
     // Receives the output of `HashToBitsGadget::hash_to_bits` in Little Endian
     // decodes the G1 point and then multiplies it by the curve's cofactor to
     // get the hash
-    fn hash_to_group<CS: ConstraintSystem<P::Fp>>(
-        mut cs: CS,
-        xof_bits: &[Boolean],
-    ) -> Result<G1Gadget<P>, SynthesisError> {
+    fn hash_to_group(
+        xof_bits: &[Boolean<Bls12_377_Fq>],
+    ) -> Result<G1Var<Bls12_377_Parameters>, SynthesisError> {
```

Comment:
I don't *really* mind, though I'm curious if this has to be specifically tied to BLS12-377.

Reply:
It doesn't, even though we only use it for 377 at present. Getting the types to work out for the more general solution is non-trivial: I would propose leaving it as-is and creating an issue to make it more general later

Reply:
Sounds good to me.